src/mumble/ConnectDialog.cpp

```diff
@@ -1828,6 +1828,7 @@ void ConnectDialog::on_qcbFilter_currentIndexChanged(int filterIndex) {
 	} else if (filter == tr("Show Populated")) {
 		g.s.ssFilter = Settings::ShowPopulated;
 	}
-
-	filterPublicServerList();
+        if (!g.s.bDisablePublicList) {
+            filterPublicServerList();
+        }
```

Comment:
I think it'd make more sense to add this check to `filterPublicServerList` itself than adding it here at this particular position.

Aka: The first thing I'd do inside `filterPublicServerList` would be to check whether the public server list is disabled and if it is, return immediately as in that case there is nothing to be filtered.
In that case you don't have to conditionally call the filter-function only when the list is enabled.

This has the advantage that if someone adds a call to the filter-function at another place in the future they can't forget to check that the list is actually enabled as that's done in the function's body.

Reply:
Agreed